catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/Harvester.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * A {@code Harvester} is responsible for fetching remote resources and notifying any registered
+ * {@link Listener}s of create, update, and delete events.
+ *
+ * <p>In the event that a {@code Harvester} goes down, it should make a best effort to remember the
+ * last state of the external server it was harvesting from. Upon a {@code Harvester} coming back
+ * up, it should notify registered {@link Listener}s of new events.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Harvester {
+
+  /**
+   * Registers a new {@link Listener} to receive events from the remote source being harvested from.
+   * Newly-registered {@link Listener}s will only receive new events from this {@code Harvester}. If
+   * a {@link Listener} is already registered, the call to this method will be a no-op.
+   *
+   * @param listener the {@link Listener} to register
+   * @throws IllegalArgumentException if the {@link Listener} cannot be registered
+   */
+  void registerListener(Listener listener);
```

Comment:
I'd prefer something like the whiteboard pattern personally. What happens if the listener comes up before the harvester?

Reply:
For the first Listener implementation, Listeners would be notified when a new Harvester was registered and any interested Listeners (based on Harvester's ID) would register themselves to the Harvester. Could certainly invert that and get rid of the register/unregister methods. One thing to note is that if we get rid of these methods the API would assume a registry is available. I don't feel strongly either way though.

Reply:
I guess my other concern is that we're not specifying all of that in the API contracts. I'd rather let the framework take care of what it can handle already. I think it will be relatively trivial in the future to refactor or replace that once we start splitting things apart.